developer_documentation/contribution_guidelines.Rmd

```diff
@@ -91,6 +92,74 @@ the devil].
  members of the group also have diverse backgrounds and likely can
  give valuable design insights.
 
+### Follow BioCro's git branching structure
+
+* In general, BioCro development follows the [git-flow][] branching model, where
+  there are two permanent branches (`main` and `develop`) and three types of
+  temporary branches (hotfixes, features, and releases).
+
+* We have a few additional rules specific to BioCro development:
+
+  - Any merge into `main` or `develop` must be done via a pull request. This
+    requirement is enforced with GitHub branch protection rules and cannot be
+    bypassed.
+
+  - All pull requests into `main` and `develop` require approval before merging.
+    This requirement is not enforced using GitHub branch protection rules, but
+    please do not unilaterally merge a pull request without any form of approval
+    from another developer. (There are two exceptions to this rule related to
+    hotfix and release branches, described below.)
+
+  - The BioCro R package and repository follow semantic versioning of the form
+    `major.minor.bug`. Hotfix branches should typically only increment the third
+    (`bug`) number. Most release branches will increment the second (`minor`)
+    number, and very rarely the first (`major`) number. Feature branches do not
+    directly change the version number.
+
+  - Branch names should be prefixed with the branch type: either `feature/`,
+    `hotfix/`, or `release/`. Hotfix and release branches should be named with
+    the new version number; for example, `hotfix/v3.0.3` or `release/v3.1.0`.
+    Feature branch names should reflect their purpose, such as
+    `feature/describe-branch-rules`.
```

Comment:
> I'm not sure from where you gleaned the convention of using `<branch type>/` prefixes. 
>
> Perhaps you extrapolated from my having named a feature branch `feature/fix-document-quantities-workflow`.

Yes, this is what happened. I am fine with using `-` instead of `/`.

> The Driessen article suggests the branch naming convention `release-1.2` (for example) for a release that is to be tagged `v1.2`, and `hotfix-1.2.1` for a hotfix branch that is to become a release tagged as `v1.2.1`. And for feature branches, he suggests a very loose convention—namely, "anything except `master`, `develop`, `release-*`, or `hotfix-*`".

Okay, we don't need to prefix feature branch names (or "bugfix" branches). However, I don't see a reason to not write out the full version number. I think doing that makes the naming convention simpler.